lib/modulesync.rb

```diff
@@ -114,7 +121,31 @@ def self.manage_module(puppet_module, module_files, module_options, defaults, op
     if options[:noop]
       Git.update_noop(module_name, options)
     elsif !options[:offline]
-      Git.update(module_name, files_to_manage, options)
+      # Git.update() returns a boolean: true if files were pushed, false if not.
+      pushed = Git.update(module_name, files_to_manage, options)
+      return nil unless pushed
+
+      if options[:pr] && GITHUB_TOKEN
```

Comment:
I'd say if the `--pr` option is given and no token is specified it should be a hard error, probably up front.